lib/run-compilation.js

```diff
@@ -19,6 +19,10 @@ module.exports = function runCompilation (options, compiler, done) {
     .then(function linterSuccess (lint) {
       var results = lint.results;
 
+      if (options.emitErrors === false) {
+        results = downgradeErrors(results);
+      }
+
       warnings = results.filter(function (file) {
         return !file.errored && file.warnings && file.warnings.length;
```

Comment:
why not just check if `emitErrors` is disabled here [and then have the errors array return empty]?

this way we don't have to go through results more than twice, and if `emitErrors` is `false` we'll only have to go through it once.